### DISCUSSION TITLE
How to Use Rendering Templates in PocketBase Event Hooks for Custom HTML Emails?

### DISCUSSION THREAD

**Discussion by jlauding** [pocketbase/pocketbase#5000] [Q&A]
Hi PocketBase Community,

I’m working on a project where I need to send custom HTML emails using event hooks in PocketBase. I’ve referred to the documentation on rendering templates, but the example provided is only for use with routing, not with event hooks.

Specifically, I want to keep my HTML email templates in a separate file and use these templates within my event hooks to send emails. Could someone provide an example or guide me on how to achieve this?

I tried the following code:

`
const html_email = $template.loadFiles(
            `${__hooks}/views/welcome.html`,
        ).render({
            "name": 'Jon Doe',
        });
`

However, I’m not sure if this is the correct approach or if I’m missing something.

Any help or direction on how to properly set up and utilize rendering templates within event hooks for sending custom HTML emails would be greatly appreciated.

Thank you!

**Comment by ganigeorgiev** [comment]
> I’ve referred to the documentation on rendering templates, but the example provided is only for use with routing, not with event hooks.

`$template.loadFiles(...).render(...)` returns a plain HTML string. It doesn't matter whether it is for an email or a route page. It should work the same way everywhere.

To send an email with the result of the rendered template you need to assign it to the `html` field of `MailerMessage`:

```js
const html = $template.loadFiles(`${__hooks}/views/welcome.html`).render({ "name": 'Jon Doe', }); 

const message = new MailerMessage({
    from: {
        address: $app.settings().meta.senderAddress,
        name:    $app.settings().meta.senderName,
    },
    to:      [{address: "TO_EMAIL"}],
    subject: "YOUR_SUBJECT...",
    html:    html,
})

$app.newMailClient().send(message)
```

**Reply by jlauding** [reply]
I tried exactly this but it doesn't work. It seems to break at $template.loadFiles.

**Reply by jlauding** [reply]
I don't see any errors in the terminal either even if it the code seems to stop at this part.

**Reply by jlauding** [reply]
[SOLVED]

I found it, there problem was in my template.
